### ISSUE TITLE
Option to always have code blocks start at column 1.

### ISSUE THREAD

**Issue by asknet** [rvben/rumdl#237]
Hi,

In Atlassian markdown markup, indented code blocks aren't supported. They always begin at column 1. Indentation within code block is fine.

Appreciate if you can support the rule and this will help AI generated Atlassian Markdown grounded. Thank you!

**Comment by rvben** [comment]
Hi @asknet, thanks for this suggestion. I also don't like indented code, to be honest. 😄

**Comment by chrisgrieser** [comment]
@asknet You can set already set [MD046](https://github.com/rvben/rumdl/blob/main/docs/md046.md) to `"fenced"` to disallow code blocks via indentation, in case that solves your issue.

@rvben To be honest, I think code blocks via indentation are so error-prone that it might be worth a thought to have MD046 use `"fenced"` as the default instead of `"consistent"`?

**Comment by rvben** [comment]
@chrisgrieser is indeed right: `style = "fenced"` should be sufficient. I checked Atlassian's handling of code blocks and didn't see anything special that would need to be added to improve support. Is `style = "fenced"` enough, or is there something that you'd like to see, @asknet?

**Comment by asknet** [comment]
thanks @chrisgrieser  @rvben . I tested this and didn't work. For the below markdown it's not throwing any lint error. Expected to throw an error as code block is indented. Using version `rumdl 0.0.203`

here's the test.md file
```
## Test

    ```js
    var foo = "hello";
    ```

```

And below is the settings in config file
```toml
#MD046
[code-block-style]
style = "fenced"

#MD048
[MD048]
style = "backtick"
```

**Comment by rvben** [comment]
Ah, that example helps! In your example it's actually both, so it's fenced and indented. While according to CommonMark the indentation should be interpreted as signifying a code block. As such, it should be getting a warning so that points to a bug.

I'm curious: How does this render in your Atlassian context?

**Comment by asknet** [comment]
AI generated Markdown sometimes generate indented fenced code blocks like in the example and in indented list items. When you convert this Markdown to ADF(atlassian document format) via scripts, the conversion will result in issue. However, this [online ADF builder](https://developer.atlassian.com/cloud/jira/platform/apis/document/playground/) sanitizes and de-indents always.

**Comment by asknet** [comment]
@rvben  Thank you for fixing this in `0.0.205` version

**Comment by rvben** [comment]
Thanks @asknet for the confirmation! 🚀
